netflix-roulette/src/pages/home/components/Catalog/Catalog.tsx

```diff
@@ -1,14 +1,29 @@
-import * as React from "react";
-import { GenresTabs, MoviesList, ResultCount, MoviesSort } from "./components";
+import React, { useState } from "react";
+import {
+  GenresTabs,
+  MoviesList,
+  ResultCount,
+  MoviesSort,
+  DeleteMovieDialog,
+} from "./components";
 import { CatalogContainer, FiltersContainer } from "./Catalog.styles";
+import { Movie } from "@utils/interface";
 
-export const Catalog = () => (
-  <CatalogContainer>
-    <FiltersContainer>
-      <GenresTabs />
-      <MoviesSort />
-    </FiltersContainer>
-    <ResultCount />
-    <MoviesList />
-  </CatalogContainer>
-);
+export const Catalog = () => {
+  const [movieToDelete, setMovieToDelete] = useState<Movie>(null);
```

Comment:
You seem to be passing `setMovieToDelete` down the component tree and relying on that even few levels deep. Redux is exactly the tool to avoid props drilling and share your state across multiple scattered component allover the component tree. So, there seems to be no good reason **not to** put this into Redux store.